### PR TITLE
#1683 - Completed application - E2E tests for application tracker

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Fix for PIR decline throwing an error while completing the process.
- Refactor application progress details to get rid of the Sonarcloud issue.
- Icon spacing adjusted (@hellolynn-tbtb is aware of and approved it).
- Renamed factory method saveFakeApplicationCOE to saveFakeApplicationDisbursements allowing it to be used for these E2E tests also.
- Removed prefix `${ClientTypeBaseRoute.Institution}` from some E2E that were wrongly added.
- E2E tests for the new endpoint added for the completed status.
ApplicationStudentsController(e2e)-getCompletedApplicationDetails
    ✓ Should throw NotFoundException when the application is not associated with the authenticated student. (390 ms)
    ✓ Should throw NotFoundException when the application is not in 'Completed' status. (75 ms)
    ✓ Should get application details when application is in 'Completed' status. (76 ms)
    ✓ Should get application details with scholastic standing change type when application has a scholastic standing 'Student did not complete program associated with. (60 ms)
    ✓ Should get application details with the most updated appeal status when the application has more than one student appeals associated with. (78 ms)

image

image

image

### CODE REVIEW

File: sources/packages/backend/apps/api/src/services/application/application.service.ts [hunk 1]
```diff
@@ -1050,6 +1049,7 @@ export class ApplicationService extends RecordDataModelService<Application> {
         .createQueryBuilder("application")
         .select([
           "application.id",
+          "currentAssessment.id",
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/application/application.service.ts:1052]: This is the PIR fix. The missing ID was making type think that a new assessment was supposed to be created.

File: sources/packages/backend/libs/test-utils/src/factories/application.ts [hunk 2]
```diff
@@ -29,3 +48,98 @@ export function createFakeApplication(relations?: {
...
+    auditUser: savedUser,
+  });
+  fakeOffering.offeringIntensity = OfferingIntensity.fullTime;
+  const savedOffering = await offeringRepo.save(fakeOffering);
+  fakeOriginalAssessment.offering = savedOffering;
+  const savedOriginalAssessment = await studentAssessmentRepo.save(
+    fakeOriginalAssessment,
+  );
+  savedApplication.currentAssessment = savedOriginalAssessment;
+  savedApplication.applicationStatus = ApplicationStatus.Enrolment;
```
Comment (ann-aot) [inline sources/packages/backend/libs/test-utils/src/factories/application.ts:143]: mm, we are already saving the status at line 95 right? then why this assignment

File: sources/packages/backend/libs/test-utils/src/factories/application.ts [hunk 3]
```diff
@@ -29,3 +48,98 @@ export function createFakeApplication(relations?: {
...
+ * - `institutionLocation` related location.
+ * - `disbursementValues` related disbursement schedules.
+ * - `student` related student.
+ * @param options additional options:
+ * - `createSecondDisbursement` if provided and true creates a second disbursement,
+ * otherwise only one disbursement will be created.
+ * @returns the created application and its dependencies including the disbursement
+ * with the confirmation of enrollment data.
+ */
+export async function saveFakeApplicationDisbursements(
```
Comment (ann-aot) [inline sources/packages/backend/libs/test-utils/src/factories/application.ts:66]: do we need to include the application status in function name too, as disbursements are created before the "enrollment" state

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+      .expect({
+        statusCode: 404,
+        message: "Application not found or not on Completed status.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should get application details when application is in 'Completed' status.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:105]: mm, instead of getting a fake disbursement application and changing the data in the test, can't we create a fakecompletedAppliaction? is there any reason for it?

File: sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const scholasticStanding = new StudentScholasticStanding();
+  scholasticStanding.application = relations?.application;
+  scholasticStanding.submittedData = {} as unknown;
+  scholasticStanding.submittedDate = new Date();
+  scholasticStanding.submittedBy = relations?.submittedBy;
+  scholasticStanding.note = null;
+  scholasticStanding.studentAssessment = relations?.studentAssessment;
+  scholasticStanding.referenceOffering = null;
+  scholasticStanding.unsuccessfulWeeks = null;
+  scholasticStanding.changeType =
```
Comment (ann-aot) [inline sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts:32]: if we want a scholastic standing with different `change type`, is the plan make this as a generic function or  will be creating scholastic standing for each change type

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create a scholastic standing and have it associated with the completed application.
+    const scholasticStanding = createFakeStudentScholasticStanding({
+      submittedBy: submittedByInstitutionUser,
+    });
+    // 'Student did not complete program' is the only'scholastic standing that does not generate an assessment.
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:155]: `only scholastic`

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create a scholastic standing and have it associated with the completed application.
+    const scholasticStanding = createFakeStudentScholasticStanding({
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:152]: `application` can be passed here itself right?

File: sources/packages/backend/libs/test-utils/src/factories/application.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/application.ts:143]: I do not see a reason to. I will remove the last one.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+        scholasticStandingChangeType:
+          StudentScholasticStandingChangeType.StudentDidNotCompleteProgram,
+      });
+  });
+
+  it(`Should get application details with the most updated appeal status when the application has more than one student appeals associated with.`, async () => {
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:182]: just to refresh my mind, when we say `most updated appeal status`.  is it the appeal that was created by the student most recently? or is it the last appeal that was accepted/declined by the ministry?

File: sources/packages/backend/libs/test-utils/src/factories/application.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/application.ts:66]: The `saveFakeApplicationDisbursements` was the already existing method `saveFakeApplicationCOE` and the intention was to avoid all the work to have a student application created with the disbursement records associated.
Including the status in the name would not be recommended, IMO, since the idea would be to use this method as a more generic one. Since it is a `saveFake` and not a `createFake` we may consider adding more parameters that would be avoiding an additional saving to happen on the tests themselves (which we have been avoiding for `createFake` methods). Thoughts?
@dheepak-aot @andrepestana-aot @guru-aot @sh16011993

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:105]: When we need an application with disbursements we will need: `createFakeApplication`, `createFakeStudentAssessment`, `createFakeDisbursementSchedule`, `createFakeEducationProgramOffering`, `createFakeUser`, `createFakeStudent`, etc., and then have all of them saved properly. When this method was introduced in the past for COE (and now renamed/moved for disbursements), the idea was to concentrate all the complexity of calling the previously mentioned `createFake` methods and coordinate the way that they must be saved.
Does it answer? If not maybe I am not getting the point and we can have a chat. Please let me know.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create approved student appeal
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    approvedAppealRequest.appealStatus = StudentAppealStatus.Approved;
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:195]: `approvedAppealRequest.appealStatus`  is already an `approved` object in line 194, then why this line

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+    const application = await saveFakeApplicationDisbursements(appDataSource, {
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create approved student appeal
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:193]: full stop

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    approvedAppealRequest.appealStatus = StudentAppealStatus.Approved;
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    approvedAppeal.submittedDate = addDays(-1);
+    // Create declined student appeal from yesterday
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:201]: full stop

File: sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts:32]: For any `createFakeSomething` I believe that the idea is to use as below.
```ts
const scholasticStanding = createFakeStudentScholasticStanding();
scholasticStanding.changeType = StudentScholasticStandingChangeType.StudentWithdrewFromProgram;
scholasticStanding.myOtherProperty = "My other value";
```
When should we create a parameter? When the parameter will do something else than a direct assignment.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create approved student appeal
+    const approvedAppealRequest = createFakeStudentAppealRequest();
+    approvedAppealRequest.appealStatus = StudentAppealStatus.Approved;
+    const approvedAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [approvedAppealRequest],
+    });
+    approvedAppeal.submittedDate = addDays(-1);
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:200]: mm, here `approved appeal` is submitted yesterday, and from the at line `201` declined appeal is also submitted yesterday, but I didn't see the logic for that in declined appeal. let me know if iam missing anything

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstDisbursement: {
+          coeStatus: firstDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+        appealStatus: StudentAppealStatus.Pending,
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:226]: okay,  so, I think it's the comment issue (line 201) https://github.com/bcgov/SIMS/pull/1832/files#r1147982083. latest is pending according to logic

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 1 above)
Comment (ann-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:1052]: mm, what was the issue in PIR?

File: sources/packages/backend/apps/api/src/services/application/application.service.ts [hunk 14]
```diff
@@ -1444,10 +1444,6 @@ export class ApplicationService extends RecordDataModelService<Application> {
         student: {
           id: studentId,
         },
-        studentScholasticStandings: {
```
Comment (ann-aot) [inline sources/packages/backend/apps/api/src/services/application/application.service.ts:1447]: why was these removed, same in line 1507

File: sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts (hunk 5 above)
Comment (ann-aot) [reply sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts:32]: oh..okay, that means there will be a "create something" with some default value, and it can we overwritten in each tests as per the requirement

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue [hunk 15]
```diff
@@ -63,9 +63,21 @@ import Assessment from "@/components/students/applicationTracker/Assessment.vue"
 import Enrolment from "@/components/students/applicationTracker/Enrolment.vue";
 import Completed from "@/components/students/applicationTracker/Completed.vue";
 
-interface ApplicationEndStatusIconDetails {
-  endStatusType?: "success" | "warning" | "error";
-  endStatusIcon?: string;
+class StatusIconDetails {
+  constructor(public statusType?: "success" | "warning" | "error") {}
+  get statusIcon(): string | undefined {
+    console.log("StatusIconDetails: ", this.statusType);
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue:69]: u missed to remove the console.log

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue [hunk 16]
```diff
@@ -63,9 +63,21 @@ import Assessment from "@/components/students/applicationTracker/Assessment.vue"
 import Enrolment from "@/components/students/applicationTracker/Enrolment.vue";
 import Completed from "@/components/students/applicationTracker/Completed.vue";
 
-interface ApplicationEndStatusIconDetails {
-  endStatusType?: "success" | "warning" | "error";
-  endStatusIcon?: string;
+class StatusIconDetails {
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue:66]: why a class? why did we go for the method in the setup?

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 7 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:155]: Changed.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 6 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts]: Fixed.

File: sources/packages/backend/libs/test-utils/src/factories/application.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/application.ts]: Removed.

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue [hunk 17]
```diff
@@ -182,10 +173,6 @@ export default defineComponent({
       () => props.applicationStatus === ApplicationStatus.Draft,
     );
 
-    const thumbColor = computed(() =>
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue:185]: why is this removed? is this a leftover?

File: sources/packages/web/src/components/students/applicationTracker/Cancelled.vue [hunk 18]
```diff
@@ -31,7 +31,7 @@ export default defineComponent({
       required: true,
     },
     cancelledDate: {
-      type: String,
+      type: Date,
```
Comment (ann-aot) [inline sources/packages/web/src/components/students/applicationTracker/Cancelled.vue:34]: 👍

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 8 above)
Comment (ann-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:182]: got the answer with submitted date

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 9 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:195]: Is this case it was just to make clear the difference between both objects below? I did not see any harm in doing it. Please let me know if this is a suggestion or a blocker.
`approvedAppealRequest.appealStatus = StudentAppealStatus.Approved;`
and
`pendingAppealRequest.appealStatus = StudentAppealStatus.Pending;`

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 10 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:193]: I will add the period.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 11 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:201]: I will add the period.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 12 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:200]: I changed it and did not update the comment. I will change the comment.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 10 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts]: Period added.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 12 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:200]: Comment updated.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 11 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:201]: Period added.

File: sources/packages/backend/apps/api/src/services/application/application.service.ts (hunk 14 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/services/application/application.service.ts:1447]: The condition was introduced on the previous PR for the same ticket and was preventing a record without a scholastic standing from being returned.

File: sources/packages/backend/apps/api/src/services/application/application.service.ts [hunk 19]
```diff
@@ -1504,10 +1500,6 @@ export class ApplicationService extends RecordDataModelService<Application> {
         student: {
           id: options?.studentId,
         },
-        studentScholasticStandings: {
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/services/application/application.service.ts:1507]: 👍

File: sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+  User,
+} from "@sims/sims-db";
+
+/**
+ * Creates a scholastic standing record ready to be saved to the database.
+ * @param relations dependencies.
+ * - `submittedBy` institution user submitting the scholastic standing change.
+ * - `application` related student application.
+ * - `studentAssessment` related assessment.
+ * - `student` related student.
```
Comment (andrepestana-aot) [inline sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts:15]: Please remove student comment.

**Review by andrepestana-aot — APPROVED** [review]
Great job! Left a minor comment.

File: sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts (hunk 20 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts]: Removed.

File: sources/packages/backend/libs/test-utils/src/factories/application.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/application.ts:66]: As per the Team decision and as per mentioned in the above comment the method `saveFakeApplicationDisbursements` is now receiving the application status in the options and adjusting other properties. It causes some minor impact along with other files but I believe that it will not disturb the PR in general.

File: sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts:109]: I guess that the answer is the same from https://github.com/bcgov/SIMS/pull/1832#discussion_r1148234270

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue (hunk 16 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue]: I believe that after the chat it was clear, either was I am changing it to avoid confusion in the future.

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue [hunk 21]
```diff
@@ -127,48 +137,28 @@ export default defineComponent({
...
-          ApplicationExceptionStatus.Declined ||
-        applicationProgressDetails.value.firstCOEStatus ===
-          COEStatus.declined ||
-        applicationProgressDetails.value.secondCOEStatus === COEStatus.declined
-      ) {
-        applicationEndStatus.value = {
-          endStatusType: "error",
-          endStatusIcon: "fa:fas fa-exclamation-circle",
-        };
+        statusIconDetails.value = new StatusIconDetails("success");
```
[the remark this answers is not in the document]
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue]: Yes, I thought the same, but after our last Team conversation I thought that it would potentially cause more questions than benefits and that is why I did a refactor here. Please let me know if it is still ok.

File: sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue [hunk 22]
```diff
@@ -127,48 +142,28 @@ export default defineComponent({
...
-          ApplicationExceptionStatus.Declined ||
-        applicationProgressDetails.value.firstCOEStatus ===
-          COEStatus.declined ||
-        applicationProgressDetails.value.secondCOEStatus === COEStatus.declined
-      ) {
-        applicationEndStatus.value = {
-          endStatusType: "error",
-          endStatusIcon: "fa:fas fa-exclamation-circle",
-        };
+        statusIconDetails.value = STATUS_ICON_SUCCESS;
```
[the remark this answers is not in the document]
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue:166]: This is the refactoring of the previous `class`. There is only one interface, I am not following what would be "avoid using 3 interfaces.".

**Comment by dheepak-aot** [comment]
Good work @andrewsignori-aot . Thanks for additional refactors. Just one minor comment.

**Review by dheepak-aot — APPROVED** [review]
LGTM. 👍

**Review by ann-aot — APPROVED** [review]
Good work @andrewsignori-aot 👍 Thanks for doing the changes

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot
